### ISSUE TITLE
scalafix can't parse new scala 3.6.2 new given synstax

### ISSUE THREAD

**Issue by lenguyenthanh** [scalacenter/scalafix#2138]
The changes is here: https://github.com/lichess-org/lila-fishnet/pull/376/commits/1163f62523bf027ebb0b9a28c6e0e4a78535ac63

And the error is below:

```
info] Running scalafix on 18 Scala sources
[error] xxx/lichess/lila-fishnet/app/src/main/scala/http/MkHttpServer.scala:19:9: error: abstract givens cannot be anonymous
[error]   given Logger[IO] => MkHttpServer = new:
[error]         ^
[error] (app / scalafixAll) scalafix.sbt.ScalafixFailed: ParseError
[error] Total time: 0 s, completed 11 Dec 2024, 7:15:23 pm
```

this pr in scalameta may fix this issue: https://github.com/scalameta/scalameta/pull/4068

**Comment by lenguyenthanh** [comment]
thanks! I tried it fixed all but one issue:

```
[error] {redacted}/lila-search/modules/app/src/main/scala/http.server.scala:20:9: error: abstract givens cannot be anonymous
[error]   given Logger[IO] => MkHttpServer = new:
[error]         ^
```

I think this is still scalameta issue, you can check the code here: https://github.com/lichess-org/lila-search/pull/392/files#diff-665a50336c22a54ed11c39141cc1649c40ec1cd141de9adc3754b95e000f7891R20

**Comment by ivan-klass** [comment]
similarly, new context bound syntax gives an error:
```
[error] MyFile.scala:34:21: error: illegal start of declaration
[error]   def apply[F[_]: { Async, Tracer, Parallel }](
[error]                     ^^^^^
```

**Comment by tgodzik** [comment]
I will take a look at those, thanks!

**Comment by tgodzik** [comment]
> def apply[F[_]: { Async, Tracer, Parallel }](

This seems to parse ok, so maybe it's just not working in the released version.

**Comment by bjaglin** [comment]
> > def apply[F[_]: { Async, Tracer, Parallel }](
> 
> This seems to parse ok, so maybe it's just not working in the released version.

@ivan-klass could you confirm it works with the[ latest snapshot](https://scalacenter.github.io/scalafix/docs/users/installation.html#snapshot)?

**Comment by bjaglin** [comment]
> thanks! I tried it fixed some but still at least some issue:
> 
> ```
> [error] {redacted}/lila-search/modules/app/src/main/scala/http.server.scala:20:9: error: abstract givens cannot be anonymous
> [error]   given Logger[IO] => MkHttpServer = new:
> [error]         ^
> ```
> 
> I think this is still scalameta issue, you can check the code here: https://github.com/lichess-org/lila-search/pull/392/files#diff-665a50336c22a54ed11c39141cc1649c40ec1cd141de9adc3754b95e000f7891R20

indeed, filed https://github.com/scalameta/scalameta/issues/4091 upstream

**Comment by bjaglin** [comment]
[Latest snapshot](https://scalacenter.github.io/scalafix/docs/users/installation.html#snapshot) with the fix is [on its way](https://github.com/scalacenter/scalafix/actions/runs/12525256873/job/34936642401). I am planning to cut a release within 2 weeks, at the same time as [Scala 2.13.16](https://contributors.scala-lang.org/t/scala-2-13-16-release-planning/6773).

**Comment by ivan-klass** [comment]
> `def apply[F[_]: { Async, Tracer, Parallel }](`
> @ivan-klass could you confirm it works with the latest snapshot?

@bjaglin yes, that worked!
> I am planning to cut a release within 2 weeks, at the same time as Scala 2.13.16

Given that scala 2.13.16 artifacts are published, we're looking forward for the stable scalafix tag 🙂

**Comment by bjaglin** [comment]
> Given that scala 2.13.16 artifacts are published

Indeed, I was just waiting for some potential outstanding issue on the freshly released scalameta thanks to scalafmt 3.8.4 exposure, but it looks good - I'll cut a release tonight.
